File: src/trackmate/adapters/telegram/keyboards.py
from aiogram.types import InlineKeyboardButton, InlineKeyboardMarkup


def setup_keyboard() -> InlineKeyboardMarkup:
    return InlineKeyboardMarkup(
        inline_keyboard=[
            [InlineKeyboardButton(text="🔄 Проверить снова", callback_data="setup:check")],
            [InlineKeyboardButton(text="✨ Оформить группу", callback_data="setup:start")],
        ]
    )


def today_control_keyboard() -> InlineKeyboardMarkup:
    return InlineKeyboardMarkup(
        inline_keyboard=[
            [InlineKeyboardButton(text="➕ Добавить задачу", callback_data="today:add")],
        ]
    )


def material_progress_keyboard(batch_id: int) -> InlineKeyboardMarkup:
    return InlineKeyboardMarkup(
        inline_keyboard=[
            [
                InlineKeyboardButton(text="👀 Прочитал", callback_data=f"material:read:{batch_id}"),
                InlineKeyboardButton(text="📝 Заметка", callback_data=f"material:note:{batch_id}"),
                InlineKeyboardButton(text="🚀 Внедрил", callback_data=f"material:applied:{batch_id}"),
            ]
        ]
    )


def daily_task_keyboard(task_id: int) -> InlineKeyboardMarkup:
    return InlineKeyboardMarkup(
        inline_keyboard=[
            [InlineKeyboardButton(text="🏁 Отчитаться", callback_data=f"task:report:{task_id}")],
        ]
    )


def daily_task_status_keyboard(task_id: int) -> InlineKeyboardMarkup:
    return InlineKeyboardMarkup(
        inline_keyboard=[
            [
                InlineKeyboardButton(text="✅ Выполнена", callback_data=f"task:status:{task_id}:done"),
                InlineKeyboardButton(text="🔸 Выполнена частично", callback_data=f"task:status:{task_id}:partial"),
                InlineKeyboardButton(text="❌ Не выполнена", callback_data=f"task:status:{task_id}:failed"),
            ]
        ]
    )


def alert_keyboard(task_id: int, alert_id: int) -> InlineKeyboardMarkup:
    return InlineKeyboardMarkup(
        inline_keyboard=[
            [InlineKeyboardButton(text="🏁 Отчитаться", callback_data=f"task:report:{task_id}")],
            [InlineKeyboardButton(text="👀 Понял", callback_data=f"alert:ack:{alert_id}")],
        ]
    )

File: internal/ui/keyboards.go
package ui

import (
	"fmt"

	"github.com/igor/trackmate/internal/messages"
	"github.com/igor/trackmate/internal/telegram"
)

func SetupKeyboard() *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{{Text: messages.Text("button.setup.check"), CallbackData: "setup:check"}},
		{{Text: messages.Text("button.setup.start"), CallbackData: "setup:start"}},
	}}
}

func TodayControlKeyboard() *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{{Text: messages.Text("button.today.add"), CallbackData: "today:add"}},
	}}
}

func RoutineControlKeyboard() *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{{Text: messages.Text("button.routine.configure"), CallbackData: "routine:configure"}},
	}}
}

func GoalsControlKeyboard() *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{{Text: messages.Text("button.goals.configure"), CallbackData: "goals:configure"}},
	}}
}

func EmptyKeyboard() *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{}}
}

func RoutineItemKeyboard(checkinID int64, itemIndex int) *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{
			{Text: messages.Text("button.routine.done"), CallbackData: fmt.Sprintf("routine:item:%d:%d:done", checkinID, itemIndex)},
			{Text: messages.Text("button.routine.partial"), CallbackData: fmt.Sprintf("routine:item:%d:%d:partial", checkinID, itemIndex)},
			{Text: messages.Text("button.routine.failed"), CallbackData: fmt.Sprintf("routine:item:%d:%d:failed", checkinID, itemIndex)},
		},
	}}
}

func GoalFinalStatusKeyboard(goalSetID int64) *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{
			{Text: messages.Text("button.goal.done"), CallbackData: fmt.Sprintf("goals:final:%d:done", goalSetID)},
			{Text: messages.Text("button.goal.partial"), CallbackData: fmt.Sprintf("goals:final:%d:partial", goalSetID)},
			{Text: messages.Text("button.goal.failed"), CallbackData: fmt.Sprintf("goals:final:%d:failed", goalSetID)},
		},
	}}
}

func DailyTaskKeyboard(taskID int64) *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{{Text: messages.Text("button.task.report"), CallbackData: fmt.Sprintf("task:report:%d", taskID)}},
	}}
}

func DailyTaskStatusKeyboard(taskID int64) *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{
			{Text: messages.Text("button.task.done"), CallbackData: fmt.Sprintf("task:status:%d:done", taskID)},
			{Text: messages.Text("button.task.partial"), CallbackData: fmt.Sprintf("task:status:%d:partial", taskID)},
			{Text: messages.Text("button.task.failed"), CallbackData: fmt.Sprintf("task:status:%d:failed", taskID)},
		},
	}}
}

func AlertKeyboard(taskID int64, alertID int64) *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{{Text: messages.Text("button.task.report"), CallbackData: fmt.Sprintf("task:report:%d", taskID)}},
		{{Text: messages.Text("button.dismiss"), CallbackData: fmt.Sprintf("alert:ack:%d", alertID)}},
	}}
}

func DismissKeyboard() *telegram.InlineKeyboardMarkup {
	return &telegram.InlineKeyboardMarkup{InlineKeyboard: [][]telegram.InlineKeyboardButton{
		{{Text: messages.Text("button.dismiss"), CallbackData: "notice:dismiss"}},
	}}
}
